Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.h

```diff
@@ -33,12 +35,16 @@ namespace Terrain
         AZ_CLASS_ALLOCATOR(TerrainPhysicsSurfaceMaterialMapping, AZ::SystemAllocator, 0);
         AZ_RTTI(TerrainPhysicsSurfaceMaterialMapping, "{A88B5289-DFCD-4564-8395-E2177DFE5B18}");
         static void Reflect(AZ::ReflectContext* context);
+        static AZ::Data::AssetId GetMaterialLibraryId();
+
+        AZStd::vector<AZStd::pair<AZ::u32, AZStd::string>> BuildSelectableTagList() const;
+        void SetTagListProvider(const EditorSurfaceTagListProvider* tagListProvider);
 
         SurfaceData::SurfaceTag m_surfaceTag;
         Physics::MaterialId m_materialId;
 
     private:
-        static AZ::Data::AssetId GetMaterialLibraryId();
+        const EditorSurfaceTagListProvider* m_tagListProvider = nullptr;
```

Comment:
Why is this editor-specific functionality in the game component instead of EditorTerrainPhysicsColliderComponent.

Reply:
Because unlike editor/game time components, the serialized configuration structures are the same for both.
At least this data is not serialized like it is in `ColliderConfiguration::m_propertyVisibilityFlags`

Something possible to address in the reflection system refactoring..

Reply:
It sounds like we're thinking the same thing - if a structure is used for both editor and game, maybe it should have a more generic name that doesn't have "Editor" in the name.